library/core/src/ffi/c_str.rs

```diff
@@ -942,6 +992,39 @@ mod verify {
         assert_eq!(bytes, &slice[..end_idx]);
         assert!(c_str.is_safe());
     }
+
+    // const unsafe fn strlen(ptr: *const c_char) -> usize
+    #[kani::proof_for_contract(super::strlen)]
+    #[kani::unwind(32)]
+    fn check_strlen_contract() {
+        const MAX_SIZE: usize = 32;
+        let mut string: [u8; MAX_SIZE] = kani::any();
+        let nul_position: usize = kani::any_where(|x| *x < MAX_SIZE);
+        string[nul_position] = 0;
+        
+        // Ensure no null bytes before nul_position
+        for i in 0..nul_position {
+            kani::assume(string[i] != 0);
+        }
+        
+        let ptr = string.as_ptr() as *const c_char;
+        
+        unsafe {
+            let len = super::strlen(ptr);
+            
+            // Length must be less than isize::MAX
+            assert!(len < isize::MAX as usize);
+            // Length must equal nul_position
+            assert_eq!(len, nul_position);
+            // no null in between
+            for i in 0..len {
+                assert!(*ptr.add(i) != 0);
+            }
```

Comment:
In the precondition you use `contains` while here you use (both for the assumption and the assertion) explicit loops. I am wondering whether there is a performance difference between those approaches? Would you mind experimenting with 1) consistently using `contains` and 2) consistently using loops?

Reply:
postcondition you mean?

Reply:
Actually true of both: you also have it in a precondition in line 439.

Reply:
Oh gotcha. That was for `from_bytes_with_nul_unchecked` . Will check and update

Reply:
Thank you for gathering this data! I'd suggest you use a uniform solution, be it `contains` or loops. I'll leave it to you to pick.